src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -190,35 +198,46 @@ private static void applyFrameBoundsHeuristics(FrameDesc a, FrameDesc b)
             // less than half the timestamp space.
             return;
         }
-
-        int min = b.getMinSeen(), max = a.getMaxSeen();
-        int snDiff = (max - min) & 0xFFFF;
-
-        if (start != -1 || end != -1)
+        else if (tsDiff >= (-(1L << 30) & 0xFFFFFFFFL))
         {
-            // We can deal with sequence number distances (mod 16) that are
-            // less than 3.
-            if (snDiff < 3 || snDiff > (-3 & 0xFFFF))
+            logger.warn("Frames that are out of order detected.");
+        }
+        else
+        {
+            int min = b.getMinSeen(), max = a.getMaxSeen();
+            int snDiff = (max - min) & 0xFFFF;
+
+            if (start != -1 || end != -1)
             {
-                if (end == -1)
+                if (snDiff == 2)
                 {
-                    a.setEnd((max + 1) & 0xFFFF);
+                    if (end == -1)
+                    {
+                        a.setEnd((max + 1) & 0xFFFF);
+                    }
+                    else
+                    {
+                        b.setStart((min - 1) & 0xFFFF);
+                    }
                 }
-
-                if (start == -1)
+                else if (snDiff < 2 || snDiff > (-3 & 0xFFFF))
                 {
-                    b.setStart((min - 1) & 0xFFFF);
+                    logger.warn("Frame corruption or packets that are out of " +
+                        "order detected.");
                 }
             }
-        }
-        else
-        {
-            // We can deal with sequence number distances (mod 16) that are
-            // less than 4.
-            if (snDiff < 4 || snDiff > (-4 & 0xFFFF))
+            else
```

Comment:
Don't we need a check for "(start != -1 && end != -1)" here (also "(start != -1 || end != -1)" should not exclude it)?